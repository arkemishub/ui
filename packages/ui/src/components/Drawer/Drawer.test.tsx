/**
 * Copyright 2023 Arkemis S.r.l.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import { render } from "@testing-library/react";
import { Drawer } from "./index";
import userEvent from "@testing-library/user-event";

const intersectionObserverMock = () => ({
  observe: () => null,
  disconnect: () => null,
});
window.IntersectionObserver = jest
  .fn()
  .mockImplementation(intersectionObserverMock);

describe("Drawer", () => {
  test("it should match snapshot", () => {
    const { asFragment } = render(<Drawer open={true} onClose={() => null} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render drawer", () => {
    const { getByTestId } = render(<Drawer open={true} onClose={() => null} />);
    expect(getByTestId("arke-drawer")).toBeInTheDocument();
  });

  test("should render right drawer", () => {
    const { getByTestId } = render(
      <Drawer open={true} onClose={() => null} position="right" />
    );
    expect(getByTestId("arke-drawer")).toHaveClass("drawer--right");
  });

  test("should render left drawer", () => {
    const { getByTestId } = render(
      <Drawer open={true} onClose={() => null} position="left" />
    );
    expect(getByTestId("arke-drawer")).toHaveClass("drawer--left");
  });

  it("should not render drawer when open prop is false", () => {
    const { getByTestId } = render(
      <Drawer open={false} onClose={() => null} />
    );
    expect(getByTestId("arke-drawer")).toHaveClass("-translate-x-full");
  });

  it("should call onClose when close button is clicked", async () => {
    const onClose = jest.fn();
    const { getByTestId } = render(<Drawer open={true} onClose={onClose} />);
    await userEvent.click(getByTestId("arke-drawer-close"));
    expect(onClose).toHaveBeenCalled();
  });

  it("should render title", () => {
    const { getByText } = render(
      <Drawer open={true} onClose={() => null} title="Title" />
    );
    expect(getByText("Title")).toBeInTheDocument();
  });

  it("should render children", () => {
    const { getByText } = render(
      <Drawer open={true} onClose={() => null} title="Title">
        Children
      </Drawer>
    );
    expect(getByText("Children")).toBeInTheDocument();
  });
});
