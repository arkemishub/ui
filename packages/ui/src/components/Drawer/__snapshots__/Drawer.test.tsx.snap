// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer it should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="drawer__overlay visible"
  />
  <div
    class="drawer drawer--left"
    data-testid="arke-drawer"
    role="presentation"
  >
    <div
      class="drawer__head"
    >
      <button
        class="btn drawer__close__button"
      >
        <svg
          class="h-4 w-4"
          data-testid="arke-drawer-close"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
